tests/integration/idmap.bats

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	# idmap mounts requires root and kernel >= 5.12
+	requires root
+	requires_kernel 5.12
+	requires_idmap_fs /tmp
```

Comment:
With `require_idmap_fs` I think we can just remove the other requires (unless requires_idmap_fs is very slow, in which case leaving the other requires will speed things up for old kernels or non-root runs)

Reply:
These doesn't represent better the requirements. Kernel patches can be backported, so testing for 5.12 is not really that good. And we don't really need root, we need CAP^_SYS_ADMIN in several userns.

IMHO testing if we can idmap seems to be exactly what we need for the test

Reply:
We talked on slack and Francis preferred to leave it as it is now. Otherwise we need to skip the tests in all cases the binary fails, which can easily cause an oversight if we break the binary to always fail.